tests/server/test_query_params.py

```diff
@@ -424,6 +424,47 @@ def test_list_has_and(self):
         expected_ids = ["mpf_1"]
         self._check_response(request, expected_ids, len(expected_ids))
 
+    def test_awkward_not_queries(self):
+        """ Test an awkward query from the spec examples. It should return all but 2 structures
+        in the test data. The test is done in three parts:
+
+            - first query the individual expressions that make up the OR,
+            - then do an empty query to get all IDs
+            - then negate the expressions and ensure that all IDs are returned except
+              those from the first queries.
+
+        """
+        expected_ids = ["mpf_3819"]
+        request = (
+            '/structures?filter=chemical_formula_descriptive="Ba2NaTi2MnRe2Si8HO26F"'
+        )
+        self._check_response(request, expected_ids, len(expected_ids))
+
+        expected_ids = ["mpf_2"]
+        request = '/structures?filter=chemical_formula_anonymous = "A2BC" AND NOT chemical_formula_descriptive = "Ac2AgPb" '
+        self._check_response(request, expected_ids, len(expected_ids))
+
+        request = "/structures"
+        unexpected_ids = ["mpf_3819", "mpf_2"]
+        expected_ids = [
+            struct["id"]
+            for struct in self.client.get(request).json()["data"]
+            if struct["id"] not in unexpected_ids
```

Comment:
It's nicer to write it out, I think :)
```suggestion
            structure["id"]
            for structure in self.client.get(request).json()["data"]
            if structure["id"] not in unexpected_ids
```